docs/docs/how-it-works/02_infra-as-code.md

```diff
@@ -4,4 +4,28 @@ sidebar_position: 3
 title: infra-as-code
 ---
 
-This package is a legacy initiative to leverage pulumi in managing twilio configuration artifacts such as taskrouter entities. This package is a functional methodology and was setup to represent a typically vanilla flex account thats just been initialized. This package has not been maintained as the intention is to move towards a similar solution using the terraform provider, however what has been setup and used before is available for anyone looking to use it in the interim. More details on how to use this can be found [here](https://www.twilio.com/blog/intro-to-infrastructure-as-code-with-twilio-part-1). Currently there are no dependencies in this package that need to be used and is here only for reference. Feel free to remove this if not utilized.
+This package contains the alternative infastructure as code solutions, right now only one is in use terraform
+
+
+-----
+
+## Terraform
+
+The terraform solution uses a combination of 
+
+- github actions artifacts to store the terraform state using an encryption key
+- twilio cli to identify known and critical sids that can be injected into the terraform definition and import existing state
+- terraform configuration to manage studio flows and task router configuration that the features depend on.
+
+
+It uses the [twilio terraform provide](https://github.com/twilio/terraform-provider-twilio) which at time of writing does not support data sources.
+
+As the solution uses github actions to store the state, the state is prone to expire after 90 days.  To avoid this a cron is setup to download and upload the artifact every 30 days.  The main ambition with this setup is to lower the barrrier for entry and get development off the ground, the persistence of the terraform provider can be moved to another storage location as desired.  Instructions on that peice still to come.
+
+When running the release pipeline it will import and generate the state based on what is on the account but only with relation to resources declared in the configuration.  Any resources not in the configuration will be ignored.
```

Comment:
A note should be added about the extra job in CI unless you're able to add the `if` condition I mentioned in the related comment.